windows/RNFS/RNFSManager.cpp

```diff
@@ -0,0 +1,1138 @@
+            StorageFolder destFolder{ co_await dest.CreateFolderAsync(item.Name(), CreationCollisionOption::OpenIfExists) };
+            copyFolderHelper(srcFolder, destFolder);
+        }
+    }
+}
+catch (...)
+{
+    co_return;
+}
+
+
+winrt::fire_and_forget RNFSManager::getFSInfo(RN::ReactPromise<RN::JSValueObject> promise) noexcept
+try
+{
+    auto localFolder{ Windows::Storage::ApplicationData::Current().LocalFolder() };
+    auto properties{ co_await localFolder.Properties().RetrievePropertiesAsync({L"System.FreeSpace", L"System.Capacity"}) };
+
+    RN::JSValueObject result;
+    result["freeSpace"] = unbox_value<uint64_t>(properties.Lookup(L"System.FreeSpace"));
+    result["totalSpace"] = unbox_value<uint64_t>(properties.Lookup(L"System.Capacity"));
+
+    promise.Resolve(result);
+}
+catch (const hresult_error& ex)
+{
+    // "Failed to retrieve file system info."
+    promise.Reject(winrt::to_string(ex.message()).c_str());
+}
+
+
+winrt::fire_and_forget RNFSManager::unlink(std::string filepath, RN::ReactPromise<void> promise) noexcept
+try
+{
+    size_t pathLength{ filepath.length() };
+
+    if (pathLength <= 0) {
+        promise.Reject("Invalid path.");
+    }
+    else {
+        bool hasTrailingSlash{ filepath[pathLength - 1] == '\\' || filepath[pathLength - 1] == '/' };
+        std::filesystem::path path(hasTrailingSlash ? filepath.substr(0, pathLength - 1) : filepath);
+        path.make_preferred();
+
+        StorageFolder folder{ co_await StorageFolder::GetFolderFromPathAsync(path.parent_path().wstring()) };
+        auto target{ co_await folder.GetItemAsync(path.filename().wstring()) };
+        co_await target.DeleteAsync();
+
+        promise.Resolve();
+    }
+}
+catch (const hresult_error& ex)
+{
+    hresult result{ ex.code() };
+    if (result == HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND)) // FileNotFoundException
+    {
+        promise.Reject(RN::ReactError{ "ENOENT", "ENOENT: no such file or directory, open " + filepath });
+    }
+    else
+    {
+        // "Failed to unlink file" 
+        promise.Reject( winrt::to_string(ex.message()).c_str() );
+    }
+}
+
+
+winrt::fire_and_forget RNFSManager::exists(std::string filepath, RN::ReactPromise<bool> promise) noexcept
+try
+{
+    size_t fileLength{ filepath.length() };
+
+    if (fileLength <= 0) {
+        promise.Resolve(false);
+    }
+    else {
+        bool hasTrailingSlash{ filepath[fileLength - 1] == '\\' || filepath[fileLength - 1] == '/' };
+        std::filesystem::path path(hasTrailingSlash ? filepath.substr(0, fileLength - 1) : filepath);
+
+        winrt::hstring directoryPath, fileName;
+        splitPath(filepath, directoryPath, fileName);
+        StorageFolder folder{ co_await StorageFolder::GetFolderFromPathAsync(directoryPath) };
+        if (fileName.size() > 0) {
+            co_await folder.GetItemAsync(fileName);
+        }
+        promise.Resolve(true);
+    }
+}
+catch (const hresult_error& ex)
```

Comment:
This catch is never called if the filepath points to a non-existent path. These are the type of exceptions in the output:
```
Exception thrown at 0x00007FFED89D4B89 (KernelBase.dll) in MyApp.exe: WinRT originate error - 0x80070002 : 'The system cannot find the file specified.'.
Exception thrown at 0x00007FFED89D4B89 in MyApp.exe: Microsoft C++ exception: winrt::hresult_error at memory location 0x0000003BE39FD5F0.
```